src/Plot.tsx

```diff
@@ -38,7 +38,7 @@ const Plot = ({plot, onThumbnailUpdate} : PlotProps) => {
         
       case "image":
         if (!plot.thumbnail) {
-          plot.thumbnail = plot.data.toString();
+          onThumbnailUpdate(plot.data.toString());
```

Comment:
Yeah, we should not update thumbnail directly since it would not call the render function. Using `onThumbnailUpdate` is the right choice.